internal/services/iothub/iothub_data_source.go

```diff
@@ -35,6 +37,8 @@ func dataSourceIotHub() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
```

Comment:
Since this is a data source this should use 
```suggestion
			"identity": commonschema.SystemAssignedUserAssignedIdentityComputed(),
```

Reply:
FIxed.